artemis-protocols/artemis-openwire-protocol/src/test/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverterTest.java

```diff
@@ -248,6 +248,7 @@ public void testBadPropertyConversion() throws Exception {
       coreMessage.putStringProperty(hdrBrokerInTime, "5678");
       coreMessage.putStringProperty(hdrCommandId, "foo");
       coreMessage.putStringProperty(hdrDroppable, "true");
+      coreMessage.putStringProperty("", "");
```

Comment:
Maybe CORE messages also should reject empty property names as OpenWire messages:
```
java.lang.IllegalArgumentException: Property name cannot be empty or null

	at org.apache.activemq.command.ActiveMQMessage.setObjectProperty(ActiveMQMessage.java:497)
	...
```

This would also avoid the error on converting to OpenWire CORE messages with empty property names.

Reply:
I'd agree either the test or the impl should be doing something else than it currently is.

The test doesnt look to do anything to check what happens to this empty prop, so it seems like it would just behave the same if the impl behaviour went back to what it was, i.e it doesnt really test the fixed issue/behaviour.

Perhaps the test could do something like verify the count of expected props to ensure no unexpected props are there beyond those its checking values for. Or as Dom suggests the prop setter impl could also throw (the test could still check there isnt anything unexpected after conversion). Though that will obviously break whatever situation allowed the position there was to be an empty prop key in the Core message to begin with.

Reply:
I updated the Jira and the test to be more clear.

Reply:
@jbertram I see your point on backward compatibility for the CORE protocol but from JMS point of view empty property names are not allowed. What would happen if a core msg with an empty property name Is received by a qpid JMS consumer or a core JMS consumer?

Reply:
Fair enough, it seems a good compromise.